test/Constraints/rdar35142121.swift

```diff
@@ -0,0 +1,13 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+func foo<T>(_ a: T) -> Int {
+  return 0
+}
+
+func foo(_ a: (Int) -> (Int)) -> Int {
+  return 42
+}
+
+let result = foo({ (a: Int) -> Int in a + 1 })
+assert(result == 42)
```

Comment:
This doesn't really need to be an executable test. You could do `-emit-sil` and match by mangled name on the actual function that's getting called.

Reply:
Thanks for the suggestion! I'll try that right away.